specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json

```diff
@@ -0,0 +1,1463 @@
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "FDQN for the master pool."
+        }
+      },
+      "required": [
+        "dnsPrefix",
+        "vmSize"
+      ],
+      "description": "Profile for the container service master."
+    },
+    "ManagedClusterAgentPoolProfile": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique name of the agent pool profile in the context of the subscription and resource group."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "maximum": 100,
+          "minimum": 1,
+          "description": "Number of agents (VMs) to host docker containers. Allowed values must be in the range of 1 to 100 (inclusive). The default value is 1. ",
+          "default": 1
+        },
+        "vmSize": {
+          "$ref": "#/definitions/ContainerServiceVMSize",
+          "description": "Size of agent VMs."
+        },
+        "osDiskSizeGB": {
+          "$ref": "#/definitions/ContainerServiceOSDisk",
+          "description": "OS Disk Size in GB to be used to specify the disk size for every machine in this master/agent pool. If you specify 0, it will apply the default osDisk size according to the vmSize specified."
+        },
+        "vnetSubnetID": {
+          "$ref": "#/definitions/ContainerServiceVnetSubnetID",
+          "description": "VNet SubnetID specifies the vnet's subnet identifier."
+        },
+        "maxPods": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Maximum number of pods that can run on a node."
+        },
+        "osType": {
+          "$ref": "#/definitions/OSType",
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        },
+        "maxCount":{
+          "type": "integer",
+          "format": "int32",
+          "description": "Maximun number of nodes for auto-scaling"
+        },
+        "minCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Minimun number of nodes for auto-scaling"
+        },
+        "enableAutoScaling":{
+          "type": "boolean",
+          "description": "Wheter to enable auto-scaler"
+        },
+        "type": {
+          "$ref": "#/definitions/AgentPoolType",
+          "description": "AgentPoolType represents types of agentpool"
+        }
+      },
+      "required": [
+        "name",
+        "vmSize",
+        "count"
+      ],
+      "description": "Profile for the container service agent pool."
+    },
+    "AgentPoolType":{
+      "type": "string",
+      "enum": [
+        "VirtualMachineScaleSets",
+        "AvailabilitySet"
+      ],
+      "default": "VirtualMachineScaleSets",
```

Comment:
A quick comment on the use of "defaults".

It's important to note that how "defaults" are handled in a generated SDK can vary from language to language.
Some languages will send the default values in the request body (regardless of whether it's set by the user) and other languages will leave these fields in the request as empty, and use the default purely to document the service behavior.

So please make sure that your service API will handle both scenarios the same way - i.e. it will treat a request with an empty field and a request with a default value exactly the same.

Reply:
From our Service Api, this field is read-only. So I will remove the "default field"

Reply:
Thanks @zqingqing1 - though it it's read-only you can mark it as "readOnly": true in the "ManagedClusterAgentPoolProfile"